app/src/org/commcare/network/CTInterceptorConfig.kt

```diff
@@ -0,0 +1,48 @@
+package org.commcare.network
+
+import com.appmattus.certificatetransparency.CTLogger
+import com.appmattus.certificatetransparency.VerificationResult
+import com.appmattus.certificatetransparency.certificateTransparencyInterceptor
+
+import okhttp3.OkHttpClient
+import org.commcare.core.network.HttpBuilderConfig
+import org.commcare.preferences.HiddenPreferences
+import org.commcare.util.LogTypes
+import org.javarosa.core.services.Logger
+
+/**
+ * Adds and removes Certificate Transparency (CT) network interceptor
+ *
+ */
+class CTInterceptorConfig:HttpBuilderConfig  {
+
+
+    override fun performCustomConfig(client: OkHttpClient.Builder): OkHttpClient.Builder {
+        if(HiddenPreferences.isCertificateTransparencyEnabled()) {
+            return client.addNetworkInterceptor(getInterceptor)
+        }
+
+        // In case there are CT Interceptors already attached
+        removeCTInterceptors(client)
+        return client
+    }
+
+    private val getInterceptor = certificateTransparencyInterceptor {
+        logger = object : CTLogger {
```

Comment:
What's the frequency of this request pattern? The example lists a failed log, but it looks like at a minimum we would also expect getting a log record (or multiple) for each channel negotiation, which I don't think makes Logger the right channel unless we're filtering for failure more specifically

Based on the noisiness of logging I've been trying to push us back from using the CommCare logs for any events which are "unexceptional" and which would occur with higher frequency than user sessions. IE: If something normal / expected happens once per login it's fine to log, but if it happens on "machine frequency" (IE: a background process is retrying over and over) we should push to LogCat rather than CommCare's logs. It's also good to log blocking *failures* (Since eventually a user recovers from them), we just don't want to be constantly filling our sumo allocations with that data indefinitely.

Reply:
Completely, indeed this would create a lot of noise as it's supposed to log the result of every negotiation. I will change it to only log failures (only once), which will allow us to know when someone gets blocked. However, I don't think in that situation we would receive anything from the device.

Reply:
Cool, thanks for confirming. It's totally possible we could get the logged data from a request later if, for example, someone is trying to trick your WIFI with a MITM temporarily but you switch to mobile internet because it's not working.